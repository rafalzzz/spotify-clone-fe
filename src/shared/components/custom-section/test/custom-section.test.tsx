import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { CustomSection } from '../index';

const TITLE = 'Test Section';
const REDIRECTION_URL = '/mocked-url';
const CHILD_TEST_ID = 'child-test-id';
const CONTENT = 'Mocked Content';

const CUSTOM_SECTION_HEADER_TEST_ID = 'custom-section-header';
const CUSTOM_SECTION_SHOW_ALL_ITEMS_TEST_ID = 'custom-section-show-all-items';

const renderCustomSection = () =>
  render(
    <CustomSection title={TITLE} redirectionUrl={REDIRECTION_URL}>
      <div data-testid={CHILD_TEST_ID}>{CONTENT}</div>
    </CustomSection>,
  );

describe('CustomSection', () => {
  it('renders component without error', () => {
    const screen = renderCustomSection();
    expect(screen).toMatchSnapshot();
  });

  it('renders the section with provided title', () => {
    const { queryByRole } = renderCustomSection();
    expect(queryByRole('heading', { name: TITLE })).toBeInTheDocument();
  });

  it('renders the children content', () => {
    const { queryByTestId } = renderCustomSection();
    expect(queryByTestId(CHILD_TEST_ID)).toHaveTextContent(CONTENT);
  });

  it('contains a link with the provided redirection URL for the title', () => {
    const { queryByTestId } = renderCustomSection();
    const titleLink = queryByTestId(CUSTOM_SECTION_HEADER_TEST_ID);
    expect(titleLink).toHaveAttribute('href', REDIRECTION_URL);
  });

  it('contains a link with the provided redirection URL for the "Show all"', () => {
    const { queryByTestId } = renderCustomSection();
    const showAllLink = queryByTestId(CUSTOM_SECTION_SHOW_ALL_ITEMS_TEST_ID);
    expect(showAllLink).toHaveAttribute('href', REDIRECTION_URL);
  });
});
