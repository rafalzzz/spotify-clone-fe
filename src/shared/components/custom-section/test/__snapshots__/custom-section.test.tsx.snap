// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSection renders component without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="custom-section"
      >
        <div
          class="custom-section__header"
        >
          <header>
            <a
              class="custom-section__header__title"
              data-testid="custom-section-header"
              href="/mocked-url"
            >
              <h2>
                Test Section
              </h2>
            </a>
          </header>
          <a
            class="custom-section__header__redirection"
            data-testid="custom-section-show-all-items"
            href="/mocked-url"
          >
            Show all
          </a>
        </div>
        <div
          data-testid="child-test-id"
        >
          Mocked Content
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="custom-section"
    >
      <div
        class="custom-section__header"
      >
        <header>
          <a
            class="custom-section__header__title"
            data-testid="custom-section-header"
            href="/mocked-url"
          >
            <h2>
              Test Section
            </h2>
          </a>
        </header>
        <a
          class="custom-section__header__redirection"
          data-testid="custom-section-show-all-items"
          href="/mocked-url"
        >
          Show all
        </a>
      </div>
      <div
        data-testid="child-test-id"
      >
        Mocked Content
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
